### ISSUE TITLE
[Feature Request]: Making adding tags to many files with different tags less tedious

### ISSUE THREAD

**Issue by Thecreatre** [TagStudioDev/TagStudio#337]
### Checklist

- [X] I am using an up-to-date version.
- [X] I have read the [documentation](https://github.com/TagStudioDev/TagStudio/blob/main/doc/index.md).
- [X] I have searched existing [issues](https://github.com/TagStudioDev/TagStudio/issues).

### Description

So as of now the way that tag studio handles doing this is it will just allow you to add tags to as many files as you like, with the exception that if anything in the tag Field is different then it just says "mixed fields" and doesn't let the user jam tags into it, this makes it annoying cause now the only thing short of more fields cluttering under the image is just  well selecting the ones that are similar in the group and adding the tag and slowly doing it that way. Which is a pain for obvious reasons.

it also gets worse when some one is using more tags for stuff, because that means there's more times you have to stop to get  a tag into that one file and then you can move on

this is less of a problem for newly created library's the but older they  get  and the more tags get, the more painful it is to add anymore  to the files

(obviously you shouldn't have too many tags on a file at once due to that 100 tag limit i heard mentioned in an  other issue, but it becomes a problem the moment there's a single other tag in a file, so it gets annoying real quickly)

### Solution

Now what  i  think should work* is having all the same tags between them shown to the user and then of  course keeping that "+" button to add tags to all the selected files (oh and "-" on the tags to delete the ones you don't want of course)

![trol](https://github.com/user-attachments/assets/48986434-e14c-451a-87ac-a2970ead8623)

(visualization) (diagram)

it also should be mentioned  that there should probably be a drop down of some sort that  lets the user check the stuff that isn't similar. (especially if they trying to messily purge a large amount  of files of a tag.)

*(in the usage of the app anyway, I'm don't know coding enough to know the amount of messing with the code would be needed in the back end to make this work)

### Alternatives

I sorta already mentioned this one for what some one  could do right now to get around the problem, but its just adding  another field and adding the tags there but the  issue with that is that it  get messy real quickly, my primary ( and really only good solution i have at the moment) i think would work better.

**Comment by CyanVoxel** [comment]
Honestly, it slipped by me that this wasn't already how tag fields worked in multiple selections. I think having the common tags shown like in your example in addition to some label signaling the presence of any additional mixed tags would make the most sense here.

> it also should be mentioned that there should probably be a drop down of some sort that lets the user check the stuff that isn't similar. (especially if they trying to messily purge a large amount of files of a tag.)

Could you give an example of what you're looking for with this? If I'm understanding correctly, then I can imagine a modal popup that lists the entries along with their non-common tags, but maybe you have something similar in mind or are talking about a different scenario. 

As a side note, I believe the "100 tag limit" you mentioned you heard is a reference to the visual limit of how many tags are displayed in the search panel when adding tags, as there's no limit to how many tags can be added to or displayed on entries/files themselves.

**Comment by Thecreatre** [comment]
> Could you give an example of what you're looking for with this? If I'm understanding correctly, then I can imagine a modal popup that lists the entries along with their non-common tags, but maybe you have something similar in mind or are talking about a different scenario.

well what i was thinking was like  either a toggle or a drop down that shows a list of  the not in common tags, but in order of how common they are (as in the number of the tag over  the selected files)

bottom would be the least usage 

and the top  would be have the most  usage (out of the ones that arnt just used over all of them)

oh and  displaying a number on/ next to the tag displaying how much its used ( weather it would be a percent i unno (maybe make that  a setting it self?))

>tag55: 54
>tag 83: 23
>tag 22: 11

(scuffed display of what i mean with out me having todo screen shot shenanigans again)

**Comment by Thecreatre** [comment]
>As a side note, I believe the "100 tag limit" you mentioned you heard is a reference to the visual limit of how many tags are displayed in the search panel when adding tags, as there's no limit to how many tags can be added to or displayed on entries/files themselves.

oh well to me it might as well be a limit sense its hard to edit stuff you cant  see (and the box can get pretty cluttered before i have even get to 50 so, i dunno about getting to 100 rn)

**Comment by CyanVoxel** [comment]
> well what i was thinking was like either a toggle or a drop down that shows a list of the not in common tags, but in order of how common they are (as in the number of the tag over the selected files)
> 
> bottom would be the least usage
> 
> and the top would be have the most usage (out of the ones that arn't just used over all of them)
> 
> oh and displaying a number on/ next to the tag displaying how much its used ( weather it would be a percent i unno (maybe make that a setting it self?))
> 
> > tag 55: 54
> > tag 83: 23
> > tag 22: 11
> 
> (scuffed display of what i mean with out me having todo screen shot shenanigans again)

I see what you're saying now, that would make it easier to quickly tell if one or a few of the files are missing a tag that aaaall the other ones have, right? Or giving a helpful glance at your other frequently used tags for those files where you can quickly click to apply that tag to the rest of those files. I actually *really* like that...

**Comment by Thecreatre** [comment]
yeah, i was kinda just describing how it could be used to delete  tags, but yeah it probably could be used to check the tags are in order too

**Comment by coolesding** [comment]
This is the same as #362 

I think this should have a high priority since it makes using the app on big, diverse libraries unpractical and frustrating. for my bocchi the rock library it would take hours of frustratingly and tediously adding tags one by one to images that have common and mismatched attributes alike.

**Comment by Thecreatre** [comment]
is there any updates on this or the other post about this?

**Comment by Tyrannicodin** [comment]
I think we can do this, but we're going to wait until #332 is merged as it changes a lot with how tags are stored. I'll add it to my todo list though.

**Comment by Thecreatre** [comment]
> I think we can do this, but we're going to wait until #332 is merged as it changes a lot with how tags are stored. I'll add it to my todo list though.

ah, cool

unrelated but i sorta assume it will exist but.. when that does drop there will be a way to convert to the new format?

i ask cause from what little of read over that pull req it doesn't seem to mention a converter of any sort

**Comment by CyanVoxel** [comment]
> ah, cool
> 
> unrelated but i sorta assume it will exist but.. when that does drop there will be a way to convert to the new format?
> 
> i ask cause from what little of read over that pull req it doesn't seem to mention a converter of any sort

#332 won't have 100% of the features complete when it gets pulled, but a converter along with full backwards compatibility will be in by the time a new release is shipped

**Comment by Thecreatre** [comment]
ah, oki

**Comment by Thecreatre** [comment]
> I think we can do this, but we're going to wait until #332 is merged as it changes a lot with how tags are stored. I'll add it to my todo list though.

of note that issue,  has  been  merged now  altho is being taken  into feature  parade (aaa cant spell)

**Comment by CyanVoxel** [comment]
> i dunno, my best idea to get this looked at again is to uh, bump it (by posting this comment)
> 
> albeit i haven't been keeping up that much anymore so no idea if the base line code is stable enuf to jam this in there at the current time (but ig its better than being completely forgotten (i suppose this _and_ its... oh younger friend #362 too ))

Thanks for the bump, this one slipped under my radar. Now that #332 has been merged and *most* of the backend work on that is done, there's going to be some more focus on cleaning up the UI interactions to squash bugs and add in missing functionality, specifically with #362 but now would be a perfect time to get this going as well.

**Comment by Thecreatre** [comment]
trying  the  newest PR4, it seems like the main part of this is half added,  however, it doesn't show you can do  it exactly (like it doesn't show  any tags while your  adding them to  the entries. its just a blank panel)
